### PR TITLE
core: add interfaces and signed data for attestation aggregation

### PR DESCRIPTION
**Pull request by dB2510**
Adds required interfaces to eth2wrap and signed data for SignedAggregateAndProof.

category: feature
ticket: #1094

### CODE REVIEW

File: core/signeddata.go [hunk 1]
```diff
@@ -747,6 +747,61 @@ func (s *SignedBeaconCommitteeSubscription) UnmarshalJSON(input []byte) error {
...
+
+// NewPartialSignedAggregateAndProof is a convenience function which returns new partially signed SignedAggregateAndProof.
+func NewPartialSignedAggregateAndProof(data *eth2p0.SignedAggregateAndProof, shareIdx int) ParSignedData {
+	return ParSignedData{
+		SignedData: NewSignedAggregateAndProof(data),
+		ShareIdx:   shareIdx,
+	}
+}
+
+// SignedAggregateAndProof wraps eth2p0.SignedAggregateAndProof which implements SignedData.
```
Comment (xenowits) [inline core/signeddata.go:763]:
```suggestion
// SignedAggregateAndProof wraps eth2p0.SignedAggregateAndProof and implements SignedData.
```

File: core/signeddata.go [hunk 2]
```diff
@@ -747,6 +747,61 @@ func (s *SignedBeaconCommitteeSubscription) UnmarshalJSON(input []byte) error {
 	return s.BeaconCommitteeSubscription.UnmarshalJSON(input)
 }
 
+// NewSignedAggregateAndProof is a convenience function which returns new signed SignedAggregateAndProof.
+func NewSignedAggregateAndProof(data *eth2p0.SignedAggregateAndProof) SignedAggregateAndProof {
+	return SignedAggregateAndProof{SignedAggregateAndProof: *data}
+}
+
+// NewPartialSignedAggregateAndProof is a convenience function which returns new partially signed SignedAggregateAndProof.
```
Comment (xenowits) [inline core/signeddata.go:755]: nit```suggestion
// NewPartialSignedAggregateAndProof is a convenience function which returns a new partially signed SignedAggregateAndProof.
```

File: core/signeddata.go [hunk 3]
```diff
@@ -747,6 +747,61 @@ func (s *SignedBeaconCommitteeSubscription) UnmarshalJSON(input []byte) error {
 	return s.BeaconCommitteeSubscription.UnmarshalJSON(input)
 }
 
+// NewSignedAggregateAndProof is a convenience function which returns new signed SignedAggregateAndProof.
```
Comment (xenowits) [inline core/signeddata.go:750]: nit:
```suggestion
// NewSignedAggregateAndProof is a convenience function which returns a new signed SignedAggregateAndProof.
```

**Review by xenowits — APPROVED** [review]

File: core/signeddata_test.go [hunk 4]
```diff
@@ -72,13 +73,28 @@ func TestSignedDataSetSignature(t *testing.T) {
...
+		},
 	}
 
 	for _, test := range tests {
 		t.Run(test.name, func(t *testing.T) {
+			var zeroSig core.Signature
 			clone, err := test.data.SetSignature(testutil.RandomCoreSignature())
 			require.NoError(t, err)
 			require.NotEqual(t, clone.Signature(), test.data.Signature())
+			require.NotEqual(t, clone.Signature(), zeroSig)
```
Comment (corverroos) [inline core/signeddata_test.go:97]: can also do `require.Empty`

**Review by corverroos — APPROVED** [review]

**Pull request merged**